### DISCUSSION TITLE
Library project targeting .net 7

### DISCUSSION THREAD

**Discussion by sam-wheat** [unoplatform/uno#10529] [Q&A] [answered]
Based on [this post](https://platform.uno/blog/uno-platform-4-6-hello-net-7/) I am hoping to target .net 7 in my library project.
I installed the Uno templates yesterday so I presume I am using latest (4.6.19.19).

Using VS 17.4.1

Per [this document](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows) I ran `uno-check --pre`

Tried to create the project with `dotnet new unolib -o UnoLibrary1` per doc reference above but got an error that the template was not found.  So I created a new project in VS using Uno Platform Library template.  .csproj looks like this:

	<Project Sdk="MSBuild.Sdk.Extras/3.0.23">
	  <PropertyGroup>
		<TargetFrameworks>net6.0-windows10.0.18362;netstandard2.0;net6.0-ios;net6.0-macos;net6.0-maccatalyst;net6.0-android</TargetFrameworks>
		
		<GenerateLibraryLayout>true</GenerateLibraryLayout>
	  </PropertyGroup>
	  <ItemGroup Condition="'$(TargetFramework)'=='net6.0-windows10.0.18362'">
		<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.1.3" />
		<FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.22000.24" />
		<FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.22000.24" />
	  </ItemGroup>
	  <ItemGroup Condition="'$(TargetFramework)'!='net6.0-windows10.0.18362'">
		<PackageReference Include="Uno.WinUI" Version="4.6.19" />
	  </ItemGroup>
	  <ItemGroup Condition="'$(TargetFramework)'!='net6.0-windows10.0.18362'">
		<Content Include="Assets\**" />
		<Page Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
		<Compile Update="**\*.xaml.cs">
		  <DependentUpon>%(Filename)</DependentUpon>
		</Compile>
	  </ItemGroup>
	  <ItemGroup>
		<UpToDateCheckInput Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
	  </ItemGroup>
	</Project>

Reference: 
https://github.com/unoplatform/uno/issues/10463
https://github.com/unoplatform/uno/discussions/10448

Sorry for my ignorance I am new to Uno this is my first project.

**Comment by jeromelaban** [comment, marked as answer]
.NET 7 library templates are not in yet, you'll need to change the `TargetFrameworks` property manually. It's as simple as replacing `net6.0` with `net7.0`.  

If you can't create using `dotnet new`, you've probably missed the installation of the templates (`dotnet new install Uno.ProjectTemplates.Dotnet` in the [first few lines of this document](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=windows))
